src/plugins/intel_cpu/tests/functional/custom/single_layer_tests/classes/string_tensor_pack.cpp

```diff
@@ -0,0 +1,130 @@
+        CPUSpecificParams cpuParams;
+        StringTensorPackSpecificParams StringTensorPackPar;
+        std::tie(basicParamsSet, cpuParams) = obj.param;
+        std::string td;
+        ElementType indicesPrecision;
+        std::tie(StringTensorPackPar, indicesPrecision, td) = basicParamsSet;
+
+        InputShape indicesShape;
+        InputShape symbolsShape;
+        std::vector<int64_t> begins;
+        std::vector<int64_t> ends;
+        std::tie(indicesShape, symbolsShape, begins, ends) = StringTensorPackPar;
+        std::ostringstream result;
+
+        result << ov::test::utils::partialShape2str({ indicesShape.first }) << "_";
+        result << "TS=";
+        result << "(";
+        for (const auto& targetShape : indicesShape.second) {
+            result << ov::test::utils::vec2str(targetShape) << "_";
+        }
+        result << ")";
+        result << "_symbolsShape=";
+        result << "(";
+        for (const auto& targetShape : symbolsShape.second) {
+            result << ov::test::utils::vec2str(targetShape) << "_";
+        }
+        result << ")";
+        result << "_indicesPrecision=" << indicesPrecision;
+        result << CPUTestsBase::getTestCaseName(cpuParams);
+
+        return result.str();
+    }
+
+void StringTensorPackLayerCPUTest::generate_inputs(const std::vector<ov::Shape>& targetInputStaticShapes) {
+        inputs.clear();
+        const auto& funcInputs = function->inputs();
+
+        ov::Tensor data_tensor;
+        const auto& dataShape = targetInputStaticShapes[2];
+        ov::test::utils::InputGenerateData in_data;
+        in_data.start_from = 0;
+        in_data.range = 10;
+        data_tensor = ov::test::utils::create_and_fill_tensor(ov::element::u8, dataShape, in_data);
+        inputs.insert({ funcInputs[0].get_node_shared_ptr(), data_tensor });
+    }
+
+void StringTensorPackLayerCPUTest::SetUp() {
+        StringTensorPackLayerTestParams basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::tie(basicParamsSet, cpuParams) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+
+        StringTensorPackSpecificParams StringTensorPackParams;
+        ElementType indicesPrecision;
+        std::tie(StringTensorPackParams, indicesPrecision, targetDevice) = basicParamsSet;
+
+        InputShape indicesShape;
+        InputShape symbolsShape;
+        std::vector<int64_t> begins;
+        std::vector<int64_t> ends;
+        std::tie(indicesShape, symbolsShape, begins, ends) = StringTensorPackParams;
+
+        init_input_shapes({indicesShape, indicesShape, symbolsShape});
+        auto symbolsParameter = std::make_shared<ov::op::v0::Parameter>(ov::element::u8, inputDynamicShapes[2]);
+        auto beginsConst = std::make_shared<ov::op::v0::Constant>(indicesPrecision, indicesShape.second.front(), begins);
+        auto endsConst = std::make_shared<ov::op::v0::Constant>(indicesPrecision, indicesShape.second.front(), ends);
```

Comment:
Are these inputs supposed to be constants only in real applications?

Reply:
They're rather supposed to be Parameters, good that you noticed. I changed them to Parameters.